### PR TITLE
Fix FRR manifest

### PR DESCRIPTION
**Pull request by fedepaol**
While fixing comments in https://github.com/metallb/metallb/pull/1014 , I cherry-picked a wrong version of the frr manifest, with the result that CI is running legacy mode even on the CI lanes (this is what we call volkswagen test https://github.com/auchenberg/volkswagen)

Here, we fix the manifest and we add a test (running only on the frr lanes) that validates the FRR container in the pods is actually working.

### CODE REVIEW

File: manifests/metallb-frr.yaml [hunk 1]
```diff
@@ -538,6 +526,8 @@ spec:
           volumeMounts:
             - name: reloader
               mountPath: /etc/frr_reloader
+            - name: frr-conf
+              mountPath: /etc/frr
```
Comment (oribon) [inline manifests/metallb-frr.yaml:530]: why is this needed?

Comment (fedepaol) [reply manifests/metallb-frr.yaml:530]: probably a leftover of me hammering through making it work.
Will remove it.

Comment (fedepaol) [reply manifests/metallb-frr.yaml]: Removed

**Comment by russellb** [comment]
seems the "make sure frr is running" test failed?

> [Fail] BGP type=Loadbalancer [BeforeEach] would check that FRR is running in the speaker /home/circleci/project/e2etest/bgp_test.go:376

**Comment by fedepaol** [comment]
> seems the "make sure frr is running" test failed?
> 
> > [Fail] BGP type=Loadbalancer [BeforeEach] would check that FRR is running in the speaker /home/circleci/project/e2etest/bgp_test.go:376

what's weird is, it failed on the part common to the loadbalancer tests https://github.com/metallb/metallb/blob/bbd0cf14a2c4eb7aaf96d4e1cacda5169fc2ce98/e2etest/bgp_test.go#L120

File: e2etest/bgp_test.go [hunk 2]
```diff
@@ -149,6 +149,11 @@ var _ = ginkgo.Describe("BGP", func() {
 
 			validateService(cs, svc, epNodes)
 		})
+
+		ginkgo.It("would check that FRR is running in the speaker", func() {
+			paired := frrIsPairedOnPods(cs, frrContainerIPv4)
+			framework.ExpectEqual(paired, true)
```
Comment (russellb) [inline e2etest/bgp_test.go:155]: could add the opposite check for non-frr test jobs (make sure the frr container isn't running when it's not supposed to be)

**Review by russellb — COMMENTED** [review]
lgtm, one idea to enhance the check, but don't need to block on it
